src/ReadableEnum.php

```diff
@@ -12,6 +12,11 @@
 
 use Elao\Enum\Exception\InvalidValueException;
 
+/**
+ * @template T of int|string
+ * @extends Enum<T>
+ * @implements ReadableEnumInterface<T>
```

Comment:
This stuff is useful when creating the enums so that `phpstan` and `psalm` will detect the return type of `getValue()`, `values()`, etc.